cmd/setup-etcd-environment/run_test.go

```diff
@@ -16,164 +138,231 @@ import (
-		t.Errorf("bootEnv[\"INITIAL_CLUSTER_STATE\"] = \"%s\"; want \"\" ", bootEnv["INITIAL_CLUSTER_STATE"])
+		t.Errorf("bootEnv[\"INITIAL_CLUSTER_STATE\"] = (%s); want (%s)", bootEnv["INITIAL_CLUSTER_STATE"], "")
 	}
 }
 
 func TestFileWithNoVariables(t *testing.T) {
-	file, err := ioutil.TempFile("/tmp/", "etcd_")
+	etcdName := "etcdname.devcluster.openshift.com"
+	file, err := ioutil.TempFile("/tmp", "run-etcd-environment.*")
 	if err != nil {
 		t.Fatalf("Creating a tempfile failed: %s", err)
 	}
 	defer os.Remove(file.Name())
 
-	bootEnv, err := setBootstrapEnv(file.Name(), "etcdname.devcluster.openshift.com", true)
+	runOpts := &opts{
+		discoverySRV: "devcluster.openshift.com",
+		ifName:       "",
+		outputFile:   file.Name(),
+		bootstrapSRV: true,
+	}
+
+	setupEnv, err := NewSetupEnv(runOpts, etcdName, "./tmp", []string{testIPAddress})
+	if err != nil {
+		t.Fatalf("setupEnv() failed: %s", err)
+	}
 
+	bootEnv, err := setupEnv.getBootstrapEnv()
 	if err != nil {
 		t.Fatalf("setBootstrapEnv() failed: %s", err)
 	}
 
-	if bootEnv["DISCOVERY_SRV"] != "etcdname.devcluster.openshift.com" {
-		t.Errorf("bootEnv[\"DISCOVERY_SRV\"] = \"%s\"; want \"etcdname.devcluster.openshift.com\" ", bootEnv["DISCOVERY_SRV"])
+	if bootEnv["DISCOVERY_SRV"] != setupEnv.opts.discoverySRV {
+		t.Errorf("bootEnv[\"DISCOVERY_SRV\"] = (%s); want (%s)", bootEnv["DISCOVERY_SRV"], setupEnv.opts.discoverySRV)
 	}
 	if bootEnv["INITIAL_CLUSTER"] != "" {
-		t.Errorf("bootEnv[\"INITIAL_CLUSTER\"] = \"%s\"; want \"\" ", bootEnv["INITIAL_CLUSTER"])
+		t.Errorf("bootEnv[\"INITIAL_CLUSTER\"] = (%s); want (%s)", bootEnv["INITIAL_CLUSTER"], "")
 	}
 	if bootEnv["INITIAL_CLUSTER_STATE"] != "" {
-		t.Errorf("bootEnv[\"INITIAL_CLUSTER_STATE\"] = \"%s\"; want \"\" ", bootEnv["INITIAL_CLUSTER_STATE"])
+		t.Errorf("bootEnv[\"INITIAL_CLUSTER_STATE\"] = (%s); want (%s)", bootEnv["INITIAL_CLUSTER_STATE"], "")
 	}
 }
 
 func TestFileWithNoVariablesFalseFlag(t *testing.T) {
-	file, err := ioutil.TempFile("/tmp/", "etcd_")
+	etcdName := "etcdname.devcluster.openshift.com"
+	file, err := ioutil.TempFile("/tmp", "run-etcd-environment.*")
 	if err != nil {
 		t.Fatalf("Creating a tempfile failed: %s", err)
 	}
 	defer os.Remove(file.Name())
 
-	bootEnv, err := setBootstrapEnv(file.Name(), "etcdname.devcluster.openshift.com", false)
+	runOpts := &opts{
+		discoverySRV: "devcluster.openshift.com",
+		ifName:       "",
+		outputFile:   file.Name(),
+		bootstrapSRV: false,
+	}
+
+	setupEnv, err := NewSetupEnv(runOpts, etcdName, "./tmp", []string{testIPAddress})
+	if err != nil {
+		t.Fatalf("setupEnv() failed: %s", err)
+	}
 
+	bootEnv, err := setupEnv.getBootstrapEnv()
 	if err != nil {
 		t.Fatalf("setBootstrapEnv() failed: %s", err)
 	}
 
 	if bootEnv["DISCOVERY_SRV"] != "" {
-		t.Errorf("bootEnv[\"DISCOVERY_SRV\"] = \"%s\"; want \"\" ", bootEnv["DISCOVERY_SRV"])
+		t.Errorf("bootEnv[\"DISCOVERY_SRV\"] = (%s); want (%s)", bootEnv["DISCOVERY_SRV"], "")
```

Comment:
nit, `%q`?  But paren-wrapping is probably safe enough too :p

Reply:
good point updated!